crates/core_arch/src/x86/avx2.rs

```diff
@@ -181,6 +183,10 @@ pub unsafe fn _mm256_alignr_epi8<const IMM8: i32>(a: __m256i, b: __m256i) -> __m
     let a = a.as_i8x32();
     let b = b.as_i8x32();
 
+    if IMM8 == 16 {
```

Comment:
Would another possible fix be to change `IMM8 > 16` to `IMM8 >= 16` above?

Reply:
No, that's not enough.